src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts

```diff
@@ -0,0 +1,71 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { useEffect, useState } from 'react'
+import { logger } from 'lib/logger'
+
+import type { SupportedFiatRampConfig } from '../config'
+import { supportedFiatRamps } from '../config'
+import type { FiatRampAsset } from '../FiatRampsCommon'
+import { FiatRampAction } from '../FiatRampsCommon'
+
+const moduleLogger = logger.child({
+  namespace: ['Modals', 'FiatRamps', 'hooks', 'useFiatRampCurrencyList'],
+})
+
+type useFiatRampByAssetIdProps = {
+  assetId: AssetId | undefined
+  action: FiatRampAction
+}
+
+export const useFiatRampByAssetId = ({ assetId, action }: useFiatRampByAssetIdProps) => {
+  const [loading, setLoading] = useState(false)
+  const [providers, setProviders] = useState<SupportedFiatRampConfig[]>([])
+
+  useEffect(() => {
+    if (!assetId) return
+    setLoading(true)
+    async function getBySellAssets() {
+      const parsedProviders = (
+        await Promise.allSettled(
+          Object.values(supportedFiatRamps).map<
+            Promise<
+              [
+                provider: SupportedFiatRampConfig,
+                buyAndSellList: [FiatRampAsset[], FiatRampAsset[]],
+              ]
+            >
+          >(async provider => {
+            return [provider, await provider.getBuyAndSellList()]
```

Comment:
this is going to fire 5 requests every time we switch assets. 

we may want to fire these requests once via an effect, store them in state, and have the hook return a filtered list when the assetId changes?

Reply:
could this be refactored in the fiat ramp abstraction stuff, to read from cache. Otherwise go out and get them?